gui/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,615 @@
+        """
+        return self._connector
+
+    @connector.setter
+    def connector(self, connector_type: Union[str, np.ndarray]):
+        self._connector = connector_type
+        self._refresh()
+
+    @property
+    def mode(self):
+        """None, str: Interactive mode
+        """
+        return self._mode
+
+    @mode.setter
+    def mode(self, mode):
+        if mode == self.mode:
+            return
+        if mode == 'pan/zoom':
+            self.cursor = 'standard'
+            self.interactive = True
+            self.help = ''
+            self.status = mode
+            self._mode = mode
+        else:
+            raise ValueError("Mode not recongnized")
+
+        self.events.mode(mode=mode)
+
+    # =========================== Napari Layer ABC methods ===================
+    @property
+    def data(self) -> np.ndarray:
+        """
+
+        :return: coordinates of line vertices
+        """
+        return self.vectors
+
+    @data.setter
+    def data(self, data: np.ndarray) -> None:
+        """
+        :param data:
+        :return:
+        """
+
+        self.vectors = data
+
+    def _get_shape(self):
+        if len(self.vectors) == 0:
+            return np.ones(self.vectors.shape, dtype=int)
+        else:
+            return np.max(self.vectors, axis=0) + 1
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self._set_view_slice(self.viewer.dims.indices)
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _set_view_slice(self, indices):
+        """Sets the view given the indices to slice with.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+        in_slice_vectors, matches = self._slice_vectors(indices)
+
+        # Display vectors if there are any in this slice
+        if len(in_slice_vectors) > 0:
+            data = np.array(in_slice_vectors) + 0.5
+        else:
+            # if no vectors in this slice send dummy data
+            data = np.empty((0, 2))
+
+        self._node.set_data(
+            data[::-1],
+            width=self.width,
+            color=self.color,
+            connect=self.connector)
+        self._need_visual_update = True
+        self._update()
+
+    def _slice_vectors(self, indices):
+        """Determines the slice of markers given the indices.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+        # Get a list of the vectors for the vectors in this slice
+
+        vectors = self.vectors
+        if len(vectors) > 0:
```

Comment:
RIght now this logic is making certain assumptions about how nD Vectors data would be formatted, but you don't really support nD data at all. I'd suggest just deleting / commenting out this whole function for now and when it comes time to make this layer type be nD we can implement something new. It will likely want to incorporate the refactoring of the dims that @royerloic is working on right now, so best to wait on this